data/functions.sql

```diff
@@ -640,21 +640,94 @@ END;
 $$ LANGUAGE plpgsql IMMUTABLE;
 
 
--- calculate the list of refs (or names) of transit routes that
--- a particular station is part of. this ends up being a very
--- complex function because it's bouncing around between the
--- nodes, ways and relations to calculate membership of various
--- sets.
+-- for the purposes of sweeping for related public transit features,
+-- "interesting" tags are ones which define relations which group public
+-- transport features, or group features together in a "site". these are:
+--   * public_transport = stop_area
+--   * public_transport = stop_area_group
+--   * type = stop_area
+--   * type = stop_area_group
+--   * type = site
+--
+CREATE OR REPLACE FUNCTION mz_is_interesting_transit_relation(
+  tags hstore)
+RETURNS BOOLEAN AS $$
+BEGIN
+  RETURN
+    tags->'public_transport' IN ('stop_area', 'stop_area_group') OR
+    tags->'type' IN ('stop_area', 'stop_area_group', 'site');
+END;
+$$ LANGUAGE plpgsql IMMUTABLE;
+
+-- extract a name for a transit route relation. this can expand comma
+-- separated lists and prefers to use the ref rather than the name.
+CREATE OR REPLACE FUNCTION mz_transit_route_name(
+  tags hstore)
+RETURNS text AS $$
+DECLARE
+  route_name text;
+BEGIN
+  SELECT TRIM(BOTH FROM "name") INTO route_name
+    FROM (
+      SELECT UNNEST(string_to_array(COALESCE(
+        -- prefer ref as it's less likely to contain the destination name
+        tags->'ref',
+        tags->'name'
```

Comment:
Not necessarily, but if `tags->'name'` doesn't exist, then it will just return NULL. The whole query should return NULL and set `route_name` to NULL:

```SQL
select trim(both from "name") from (
  select unnest(string_to_array(coalesce(null::text, null::text), ',')) as "name"
) x;
```

Returns 0 rows.